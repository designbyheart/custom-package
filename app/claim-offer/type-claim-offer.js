// @flow
import type { CustomError, ResetAction } from '../common/type-common'
import type {
  AdditionalDataPayload,
  ClaimProofNavigation,
  Attribute,
  NotificationPayload,
  NotificationPayloadInfo,
} from '../push-notification/type-push-notification'

export const CLAIM_OFFER_STATUS = {
  IDLE: 'IDLE',
  RECEIVED: 'RECEIVED',
  SHOWN: 'SHOWN',
  ACCEPTED: 'ACCEPTED',
  IGNORED: 'IGNORED',
  REJECTED: 'REJECTED',
}

export const CLAIM_REQUEST_STATUS = {
  NONE: 'NONE',
  SENDING_CLAIM_REQUEST: 'SENDING_CLAIM_REQUEST',
  CLAIM_REQUEST_FAIL: 'CLAIM_REQUEST_FAIL',
  CLAIM_REQUEST_SUCCESS: 'CLAIM_REQUEST_SUCCESS',
}

export type ClaimOfferStatus = $Keys<typeof CLAIM_OFFER_STATUS>
export type ClaimRequestStatus = $Keys<typeof CLAIM_REQUEST_STATUS>

export const CLAIM_OFFER_RECEIVED = 'CLAIM_OFFER_RECEIVED'
export type ClaimOfferReceivedAction = {
  type: typeof CLAIM_OFFER_RECEIVED,
  payload: AdditionalDataPayload,
  payloadInfo: NotificationPayloadInfo,
}

export const CLAIM_OFFER_FAILED = 'CLAIM_OFFER_FAILED'
export type ClaimOfferFailedAction = {
  type: typeof CLAIM_OFFER_FAILED,
  error: CustomError,
  uid: string,
}

export const CLAIM_OFFER_SHOWN = 'CLAIM_OFFER_SHOWN'
export type ClaimOfferShownAction = {
  type: typeof CLAIM_OFFER_SHOWN,
  uid: string,
}

export const CLAIM_OFFER_ACCEPTED = 'CLAIM_OFFER_ACCEPTED'
export type ClaimOfferAcceptedAction = {
  type: typeof CLAIM_OFFER_ACCEPTED,
  uid: string,
}

export const CLAIM_OFFER_REJECTED = 'CLAIM_OFFER_REJECTED'
export type ClaimOfferRejectedAction = {
  type: typeof CLAIM_OFFER_REJECTED,
  uid: string,
}

export const CLAIM_OFFER_IGNORED = 'CLAIM_OFFER_IGNORED'
export type ClaimOfferIgnoredAction = {
  type: typeof CLAIM_OFFER_IGNORED,
  uid: string,
}

export const SEND_CLAIM_REQUEST = 'SEND_CLAIM_REQUEST'
export type SendClaimRequestAction = {
  type: typeof SEND_CLAIM_REQUEST,
  uid: string,
  payload: ClaimOfferPayload,
}

export const CLAIM_REQUEST_SUCCESS = 'CLAIM_REQUEST_SUCCESS'
export type ClaimRequestSuccessAction = {
  type: typeof CLAIM_REQUEST_SUCCESS,
  uid: string,
}

export const CLAIM_REQUEST_FAIL = 'CLAIM_REQUEST_FAIL'
export type ClaimRequestFailAction = {
  type: typeof CLAIM_REQUEST_FAIL,
  uid: string,
}

export const INITIAL_ACTION = 'INITIAL_ACTION'
export type ClaimRequestInitialAction = {
  type: typeof INITIAL_ACTION,
}

export type ClaimOfferAction =
  | ClaimOfferReceivedAction
  | ClaimOfferFailedAction
  | ClaimOfferShownAction
  | ClaimOfferAcceptedAction
  | ClaimOfferRejectedAction
  | SendClaimRequestAction
  | ClaimRequestSuccessAction
  | ClaimRequestFailAction
  | ClaimRequestInitialAction
  | ResetAction

// Assumption is that paid claim will have payTokenValue in ClaimOfferPayload/claimOfferData. CO-1329
export type ClaimOfferPayload = AdditionalDataPayload & {
  uid: string,
  senderLogoUrl?: ?string,
  remotePairwiseDID: string,
  status: ClaimOfferStatus,
  claimRequestStatus: ClaimRequestStatus,
  payTokenValue?: ?string,
}

export type ClaimOfferStore = {
  +[string]: ClaimOfferPayload,
}

export type ClaimOfferProps = {
  claimOfferShown: (uid: string) => void,
  acceptClaimOffer: (uid: string) => void,
  claimOfferRejected: (uid: string) => void,
  claimOfferIgnored: (uid: string) => void,
  updateStatusBarTheme: (color?: string) => void,
  navigation: ClaimProofNavigation,
  uid: string,
  claimOfferData: ClaimOfferPayload,
  isValid: boolean,
  logoUrl?: string,
  claimThemePrimary: string,
  claimThemeSecondary: string,
}

export type ClaimOfferState = {
  disableAcceptButton: boolean,
}

export type ClaimRequestStatusModalProps = {
  claimRequestStatus: ClaimRequestStatus,
  payload: ClaimOfferPayload,
  onContinue: () => void,
  senderLogoUrl?: string,
  isPending?: boolean,
  message1: string,
  message3: string,
  message5?: string,
}

export type ClaimRequestStatusModalState = {
  isVisible: boolean,
}

export type ClaimOfferResponse = {
  msgs: [
    {
      statusCode: string,
      edgeAgentPayload: string,
      typ: string,
      statusMsg: string,
      uid: string,
    },
  ],
}

export type ClaimOfferAttributeListProps = {
  list: Array<Attribute>,
}
