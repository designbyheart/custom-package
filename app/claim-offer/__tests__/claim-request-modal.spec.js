// @flow
import React from 'react'
import 'react-native'
import renderer from 'react-test-renderer'
import ClaimRequestModal from '../claim-request-modal'
import { CLAIM_REQUEST_STATUS } from '../type-claim-offer'

describe('<ClaimRequestModal />', () => {
  it('should match snapshot', () => {
    const payload = {
      claimOffer: {
        name: 'Home Address',
        version: '1.0.0',
        revealedAttributes: [
          {
            label: 'Address 1',
            data: 'An Address',
          },
          {
            label: 'Address 2',
            data: 'An Address 2',
          },
        ],
      },
      issuer: {
        name: 'Test Issuer',
        logoUrl: 'http://testissuer.com/logoUrl.png',
        pairwiseDID: 'ha66899sadfjZJGINKN0770',
      },
    }
    const onContinue = jest.fn()
    const wrapper = renderer
      .create(
        <ClaimRequestModal
          claimRequestStatus={CLAIM_REQUEST_STATUS.SENDING_CLAIM_REQUEST}
          payload={payload}
          onContinue={onContinue}
        />
      )
      .toJSON()
    expect(wrapper).toMatchSnapshot()
  })
})
