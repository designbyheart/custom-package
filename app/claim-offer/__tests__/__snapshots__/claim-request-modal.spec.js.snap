// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ClaimRequestModal /> should match snapshot 1`] = `
<Modal
  animationIn="zoomIn"
  animationOut="zoomOut"
  animationOutTiming={100}
  backdropColor="#ffffff"
  backdropOpacity={1}
  isVisible={true}
>
  <View
    fifth={true}
    shadow={true}
    style={
      Object {
        "backgroundColor": "#ffffff",
        "marginHorizontal": 10,
        "shadowColor": "rgba(0, 0, 0, 0.2)",
        "shadowOffset": Object {
          "height": 10,
          "width": 0,
        },
        "shadowOpacity": 1,
        "shadowRadius": 30,
      }
    }
  >
    <View
      spaceBetween={true}
      style={
        Object {
          "borderBottomColor": "#3f3f3f",
          "borderBottomWidth": 0.2,
          "justifyContent": "space-between",
          "paddingVertical": 10,
        }
      }
    >
      <View
        row={true}
        spaceBetween={true}
        style={
          Object {
            "alignItems": "center",
            "flexDirection": "row",
            "justifyContent": "space-between",
            "marginHorizontal": 10,
            "marginVertical": 10,
          }
        }
        testID="claim-request-text-avatars-container"
        vCenter={true}
      >
        <View
          accessibilityComponentType={undefined}
          accessibilityLabel={undefined}
          accessibilityTraits={undefined}
          accessible={true}
          hitSlop={undefined}
          nativeID={undefined}
          onLayout={undefined}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          pointerEvents="box-only"
          style={undefined}
          testID={undefined}
        >
          <View
            imageShadow="imageShadow"
            style={
              Object {
                "borderRadius": 38,
                "elevation": 3,
                "shadowColor": "rgba(0, 0, 0, 0.25)",
                "shadowOffset": Object {
                  "height": 3,
                  "width": 0,
                },
                "shadowOpacity": 1,
                "shadowRadius": 8,
              }
            }
            testID="avatars-pair-avatars-invitee"
          >
            <Image
              accessibilityComponentType={undefined}
              accessibilityLabel={undefined}
              accessibilityTraits={undefined}
              accessible={true}
              hitSlop={undefined}
              nativeID={undefined}
              onError={undefined}
              onLayout={undefined}
              onLoad={undefined}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              source={
                Object {
                  "uri": "/var/application/DocumentDir/user-avatar.jpeg",
                }
              }
              style={
                Array [
                  Object {
                    "height": 76,
                    "width": 76,
                  },
                  Object {
                    "borderRadius": 38,
                  },
                ]
              }
              testID={undefined}
            />
          </View>
        </View>
        <Image
          source={1}
          style={
            Array [
              Object {
                "height": 22,
                "width": 30,
              },
              null,
            ]
          }
          testID="claim-request-check-mark"
        />
        <View
          imageShadow="imageShadow"
          style={
            Object {
              "borderRadius": 38,
              "elevation": 3,
              "shadowColor": "rgba(0, 0, 0, 0.25)",
              "shadowOffset": Object {
                "height": 3,
                "width": 0,
              },
              "shadowOpacity": 1,
              "shadowRadius": 8,
            }
          }
          testID="claim-request-text-avatars-inviter"
        >
          <Image
            accessibilityComponentType={undefined}
            accessibilityLabel={undefined}
            accessibilityTraits={undefined}
            accessible={true}
            hitSlop={undefined}
            nativeID={undefined}
            onError={undefined}
            onLayout={undefined}
            onLoad={undefined}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            source={
              Object {
                "uri": "http://testissuer.com/logoUrl.png",
              }
            }
            style={
              Array [
                Object {
                  "height": 76,
                  "width": 76,
                },
                Object {
                  "borderRadius": 38,
                },
              ]
            }
            testID={undefined}
          />
        </View>
      </View>
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Array [
            Object {
              "fontSize": 17,
            },
            Object {
              "fontFamily": "SF Pro Display",
            },
            Object {
              "backgroundColor": "#ffffff",
              "color": "#777777",
            },
            null,
            Object {
              "textAlign": "center",
            },
            Object {
              "backgroundColor": "transparent",
            },
            null,
            null,
            null,
            null,
            null,
            Object {
              "marginBottom": 5,
            },
          ]
        }
        testID="claim-request-message"
      >
        You accepted 
      </Text>
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Array [
            Object {
              "fontSize": 17,
            },
            Object {
              "fontFamily": "SF Pro Display",
            },
            Object {
              "backgroundColor": "#ffffff",
              "color": "#777777",
            },
            Object {
              "fontWeight": "bold",
            },
            Object {
              "textAlign": "center",
            },
            Object {
              "backgroundColor": "transparent",
            },
            null,
            null,
            null,
            null,
            null,
            Object {
              "marginBottom": 5,
            },
          ]
        }
        testID="claim-request-message"
      >
        Home Address
      </Text>
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Array [
            Object {
              "fontSize": 17,
            },
            Object {
              "fontFamily": "SF Pro Display",
            },
            Object {
              "backgroundColor": "#ffffff",
              "color": "#777777",
            },
            null,
            Object {
              "textAlign": "center",
            },
            Object {
              "backgroundColor": "transparent",
            },
            null,
            null,
            null,
            null,
            null,
            Object {
              "marginBottom": 5,
            },
          ]
        }
        testID="claim-request-message"
      >
         from 
         
      </Text>
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Array [
            Object {
              "fontSize": 17,
            },
            Object {
              "fontFamily": "SF Pro Display",
            },
            Object {
              "backgroundColor": "#ffffff",
              "color": "#777777",
            },
            Object {
              "fontWeight": "bold",
            },
            Object {
              "textAlign": "center",
            },
            Object {
              "backgroundColor": "transparent",
            },
            null,
            null,
            null,
            null,
            null,
            Object {
              "marginBottom": 5,
            },
          ]
        }
        testID="claim-request-message"
      >
        Test Issuer
      </Text>
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Array [
            Object {
              "fontSize": 17,
            },
            Object {
              "fontFamily": "SF Pro Display",
            },
            Object {
              "backgroundColor": "#ffffff",
              "color": "#777777",
            },
            null,
            Object {
              "textAlign": "center",
            },
            Object {
              "backgroundColor": "transparent",
            },
            null,
            null,
            null,
            null,
            null,
            Object {
              "marginBottom": 5,
            },
          ]
        }
        testID="claim-request-message"
      >
        !
      </Text>
    </View>
    <View
      style={
        Array [
          Object {
            "marginLeft": 15,
            "marginRight": 15,
          },
          undefined,
          Object {
            "marginLeft": 0,
            "marginRight": 0,
          },
          undefined,
        ]
      }
    >
      <View
        accessibilityComponentType={undefined}
        accessibilityLabel="Continue to see your new connection"
        accessibilityTraits={undefined}
        accessible={true}
        hasTVPreferredFocus={undefined}
        hitSlop={undefined}
        isTVSelectable={true}
        nativeID={undefined}
        onLayout={undefined}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Array [
            Object {
              "backgroundColor": "transparent",
            },
            undefined,
          ]
        }
        testID="claim-request-success-continue"
        tvParallaxProperties={undefined}
      >
        <View
          pointerEvents="box-only"
          style={
            Array [
              Object {
                "alignItems": "center",
                "backgroundColor": "#9E9E9E",
                "flexDirection": "row",
                "justifyContent": "center",
                "padding": 19,
              },
              undefined,
              undefined,
              undefined,
              undefined,
              undefined,
              Object {
                "backgroundColor": "#ffffff",
              },
              undefined,
              Object {
                "padding": 12,
              },
              undefined,
              undefined,
              undefined,
              Array [
                null,
              ],
              undefined,
              undefined,
            ]
          }
        >
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Array [
                Object {},
                undefined,
                undefined,
                undefined,
                undefined,
                undefined,
                undefined,
                undefined,
                undefined,
                undefined,
                Array [
                  Object {
                    "color": "white",
                    "fontSize": 16,
                  },
                  Object {
                    "color": "#85bf43",
                  },
                  Object {
                    "fontSize": 14,
                  },
                  undefined,
                  Object {
                    "fontWeight": "bold",
                  },
                  undefined,
                  undefined,
                  undefined,
                ],
              ]
            }
          >
            Continue
          </Text>
        </View>
      </View>
    </View>
  </View>
</Modal>
`;

exports[`<ClaimRequestModal /> should match snapshot when pending popOver 1`] = `
<Modal
  animationIn="zoomIn"
  animationOut="zoomOut"
  animationOutTiming={100}
  backdropColor="#ffffff"
  backdropOpacity={1}
  isVisible={true}
>
  <View
    fifth={true}
    shadow={true}
    style={
      Object {
        "backgroundColor": "#ffffff",
        "marginHorizontal": 10,
        "shadowColor": "rgba(0, 0, 0, 0.2)",
        "shadowOffset": Object {
          "height": 10,
          "width": 0,
        },
        "shadowOpacity": 1,
        "shadowRadius": 30,
      }
    }
  >
    <View
      spaceBetween={true}
      style={
        Object {
          "borderBottomColor": "#3f3f3f",
          "borderBottomWidth": 0.2,
          "justifyContent": "space-between",
          "paddingVertical": 10,
        }
      }
    >
      <View
        row={true}
        spaceBetween={true}
        style={
          Object {
            "alignItems": "center",
            "flexDirection": "row",
            "justifyContent": "space-between",
            "marginHorizontal": 10,
            "marginVertical": 10,
          }
        }
        testID="claim-request-text-avatars-container"
        vCenter={true}
      >
        <View
          accessibilityComponentType={undefined}
          accessibilityLabel={undefined}
          accessibilityTraits={undefined}
          accessible={true}
          hitSlop={undefined}
          nativeID={undefined}
          onLayout={undefined}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          pointerEvents="box-only"
          style={undefined}
          testID={undefined}
        >
          <View
            imageShadow="imageShadow"
            style={
              Object {
                "borderRadius": 38,
                "elevation": 3,
                "shadowColor": "rgba(0, 0, 0, 0.25)",
                "shadowOffset": Object {
                  "height": 3,
                  "width": 0,
                },
                "shadowOpacity": 1,
                "shadowRadius": 8,
              }
            }
            testID="avatars-pair-avatars-invitee"
          >
            <Image
              accessibilityComponentType={undefined}
              accessibilityLabel={undefined}
              accessibilityTraits={undefined}
              accessible={true}
              hitSlop={undefined}
              nativeID={undefined}
              onError={undefined}
              onLayout={undefined}
              onLoad={undefined}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              source={
                Object {
                  "uri": "/var/application/DocumentDir/user-avatar.jpeg",
                }
              }
              style={
                Array [
                  Object {
                    "height": 76,
                    "width": 76,
                  },
                  Object {
                    "borderRadius": 38,
                  },
                ]
              }
              testID={undefined}
            />
          </View>
        </View>
        <Image
          source={1}
          style={
            Array [
              Object {
                "height": 22,
                "width": 30,
              },
              Object {
                "height": 20,
                "right": 7,
                "width": 80,
                "zIndex": -1,
              },
            ]
          }
          testID="claim-request-check-mark"
        />
        <View
          imageShadow="imageShadow"
          style={
            Object {
              "borderRadius": 38,
              "elevation": 3,
              "shadowColor": "rgba(0, 0, 0, 0.25)",
              "shadowOffset": Object {
                "height": 3,
                "width": 0,
              },
              "shadowOpacity": 1,
              "shadowRadius": 8,
            }
          }
          testID="claim-request-text-avatars-inviter"
        >
          <Image
            accessibilityComponentType={undefined}
            accessibilityLabel={undefined}
            accessibilityTraits={undefined}
            accessible={true}
            hitSlop={undefined}
            nativeID={undefined}
            onError={undefined}
            onLayout={undefined}
            onLoad={undefined}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            source={
              Object {
                "uri": "http://testissuer.com/logoUrl.png",
              }
            }
            style={
              Array [
                Object {
                  "height": 76,
                  "width": 76,
                },
                Object {
                  "borderRadius": 38,
                },
              ]
            }
            testID={undefined}
          />
        </View>
      </View>
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Array [
            Object {
              "fontSize": 17,
            },
            Object {
              "fontFamily": "SF Pro Display",
            },
            Object {
              "backgroundColor": "#ffffff",
              "color": "#777777",
            },
            null,
            Object {
              "textAlign": "center",
            },
            Object {
              "backgroundColor": "transparent",
            },
            null,
            null,
            null,
            null,
            null,
            Object {
              "marginBottom": 5,
            },
          ]
        }
        testID="claim-request-message"
      >
        As soon as 
      </Text>
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Array [
            Object {
              "fontSize": 17,
            },
            Object {
              "fontFamily": "SF Pro Display",
            },
            Object {
              "backgroundColor": "#ffffff",
              "color": "#777777",
            },
            Object {
              "fontWeight": "bold",
            },
            Object {
              "textAlign": "center",
            },
            Object {
              "backgroundColor": "transparent",
            },
            null,
            null,
            null,
            null,
            null,
            Object {
              "marginBottom": 5,
            },
          ]
        }
        testID="claim-request-message"
      >
        Test Issuer
      </Text>
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Array [
            Object {
              "fontSize": 17,
            },
            Object {
              "fontFamily": "SF Pro Display",
            },
            Object {
              "backgroundColor": "#ffffff",
              "color": "#777777",
            },
            null,
            Object {
              "textAlign": "center",
            },
            Object {
              "backgroundColor": "transparent",
            },
            null,
            null,
            null,
            null,
            null,
            Object {
              "marginBottom": 5,
            },
          ]
        }
        testID="claim-request-message"
      >
         signs and issues 
         
      </Text>
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Array [
            Object {
              "fontSize": 17,
            },
            Object {
              "fontFamily": "SF Pro Display",
            },
            Object {
              "backgroundColor": "#ffffff",
              "color": "#777777",
            },
            Object {
              "fontWeight": "bold",
            },
            Object {
              "textAlign": "center",
            },
            Object {
              "backgroundColor": "transparent",
            },
            null,
            null,
            null,
            null,
            null,
            Object {
              "marginBottom": 5,
            },
          ]
        }
        testID="claim-request-message"
      >
        Home Address
      </Text>
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Array [
            Object {
              "fontSize": 17,
            },
            Object {
              "fontFamily": "SF Pro Display",
            },
            Object {
              "backgroundColor": "#ffffff",
              "color": "#777777",
            },
            null,
            Object {
              "textAlign": "center",
            },
            Object {
              "backgroundColor": "transparent",
            },
            null,
            null,
            null,
            null,
            null,
            Object {
              "marginBottom": 5,
            },
          ]
        }
        testID="claim-request-message"
      >
         to you it will appear here
      </Text>
    </View>
    <View
      style={
        Array [
          Object {
            "marginLeft": 15,
            "marginRight": 15,
          },
          undefined,
          Object {
            "marginLeft": 0,
            "marginRight": 0,
          },
          undefined,
        ]
      }
    >
      <View
        accessibilityComponentType={undefined}
        accessibilityLabel="Continue to see your new connection"
        accessibilityTraits={undefined}
        accessible={true}
        hasTVPreferredFocus={undefined}
        hitSlop={undefined}
        isTVSelectable={true}
        nativeID={undefined}
        onLayout={undefined}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Array [
            Object {
              "backgroundColor": "transparent",
            },
            undefined,
          ]
        }
        testID="claim-request-success-continue"
        tvParallaxProperties={undefined}
      >
        <View
          pointerEvents="box-only"
          style={
            Array [
              Object {
                "alignItems": "center",
                "backgroundColor": "#9E9E9E",
                "flexDirection": "row",
                "justifyContent": "center",
                "padding": 19,
              },
              undefined,
              undefined,
              undefined,
              undefined,
              undefined,
              Object {
                "backgroundColor": "#ffffff",
              },
              undefined,
              Object {
                "padding": 12,
              },
              undefined,
              undefined,
              undefined,
              Array [
                null,
              ],
              undefined,
              undefined,
            ]
          }
        >
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Array [
                Object {},
                undefined,
                undefined,
                undefined,
                undefined,
                undefined,
                undefined,
                undefined,
                undefined,
                undefined,
                Array [
                  Object {
                    "color": "white",
                    "fontSize": 16,
                  },
                  Object {
                    "color": "#85bf43",
                  },
                  Object {
                    "fontSize": 14,
                  },
                  undefined,
                  Object {
                    "fontWeight": "bold",
                  },
                  undefined,
                  undefined,
                  undefined,
                ],
              ]
            }
          >
            Continue
          </Text>
        </View>
      </View>
    </View>
  </View>
</Modal>
`;
